### PR TITLE
Fix printing of extensible variants

### PR DESCRIPTION
**Pull request by IwanKaramazow**
The printer defaulted to the default (Ocaml) printer when pretty printing extensible variants.
See https://github.com/facebook/reason/issues/918

This PR tries to fix the problem & intends to make Extensible Variants great again.

### CODE REVIEW

File: src/reason_pprint_ast.ml [hunk 1]
```diff
@@ -2446,6 +2446,62 @@ class printer  ()= object(self:'self)
...
+      | Pext_rebind rebind ->
+        let r = self#longident_loc rebind in
+        let prepend = (atom "=") in
+        ([makeList ~postSpace:true [prepend; r]], None)
+    in
+    let everything = match resolved with
+      | ([], None) -> barName
+      | ([], Some res) -> add_bar sourceMappedName res
+      | (_::_ as args, None) -> add_bar nameOf (normalize args)
+      | (_::_ as args, Some res) -> add_bar nameOf (normalize (args@[res]))
```
Comment (yunxing) [inline src/reason_pprint_ast.ml]: I'm a bit confused here. What's an example of this case?

Comment (IwanKaramazow) [inline src/reason_pprint_ast.ml]: The first element of the tuple represents constructor arguments, the second an optional formatted gadt.

```ocaml
(* Case 1: No constructor arguments, neither a gadt *)
type attr = ..;
type attr += | Str

(* Case 2: No constructor arguments, is a gadt  *)
type attr = ..;
type attr += | Str :attr

(* Case 3: Has Constructor args, not a gadt  *)
type attr  = ..;
type attr += | Str string;
type attr += | Point int int;

(* Case 4: Has Constructor args & is a gadt *)
type attr  = ..;
type attr += | Point int int :attr;
```

Comment (jordwalke) [inline src/reason_pprint_ast.ml]: That would make an excellent comment!

**Comment by SanderSpies** [comment]
Within the [404](https://github.com/facebook/reason/pull/907/files#diff-7dce44be99776d19af8c2d7b0d14e2c7R5843) branch the method `type_extension` is also implemented. I'm not sure if it solves this problem, but might be worth a look.

**Comment by yunxing** [comment]
Looks good to me, merging.

**Pull request merged**